File: examples/corpus/go/03_cache.before.go
package main

import "encoding/json"

type CacheManager struct {
	store map[string]interface{}
}

func (c CacheManager) GetCopy(key string) interface{} {
	// we need a deep copy
	cp, _ := json.Marshal(c.store[key])
	var out interface{}
	json.Unmarshal(cp, &out)
	return out
}

func mergeData(data, extra map[string]interface{}) map[string]interface{} {
	tmp := map[string]interface{}{}
	for k, v := range data {
		tmp[k] = v
	}
	for k, v := range extra {
		tmp[k] = v
	}
	return tmp
}

File: examples/corpus/ts/03_cache.before.ts
class CacheManager {
  private store: Record<string, any> = {};
  getCopy(key: string): any {
    // we need a deep copy to be safe
    return JSON.parse(JSON.stringify(this.store[key]));
  }
}

function mergeData(data: Record<string, any>, extra: Record<string, any>): Record<string, any> {
  const tmp = { ...data };
  Object.assign(tmp, extra);
  return tmp;
}
